daemon/sdnotify_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+)
+
+// TestSdNotify
+func TestSdNotify(t *testing.T) {
+	testDir, e := ioutil.TempDir("/tmp/", "test-")
+	if e != nil {
+		panic(e)
+	}
+	defer os.RemoveAll(testDir)
+
+	notifySocket := testDir + "/notify-socket.sock"
+	laddr := net.UnixAddr{
+		Name: notifySocket,
+		Net:  "unixgram",
+	}
+	_, e = net.ListenUnixgram("unixgram", &laddr)
+	if e != nil {
+		panic(e)
+	}
+
+	// (true, nil) - notification supported, data has been sent
+	e = os.Setenv("NOTIFY_SOCKET", notifySocket)
+	if e != nil {
+		panic(e)
+	}
+	sent, err := SdNotify("TEST: notification supported, data sent")
+	if !sent || err != nil {
+		t.Errorf("TEST: \"notification supported, data sent\" FAILED")
```

Comment:
I think https://github.com/coreos/go-systemd/pull/185#issuecomment-232965592 meant that this message can be constructed by re-using the test string above it. Same for the other two below.